e2e/testsuite/testsuite.go

```diff
@@ -0,0 +1,286 @@
+
+	pathName := fmt.Sprintf("%s-path", s.T().Name())
+	pathName = strings.ReplaceAll(pathName, "/", "-")
+
+	ic := ibctest.NewInterchain().
+		AddChain(srcChain).
+		AddChain(dstChain).
+		AddRelayer(r, "r").
+		AddLink(ibctest.InterchainLink{
+			Chain1:  srcChain,
+			Chain2:  dstChain,
+			Relayer: r,
+			Path:    pathName,
+		})
+
+	channelOptions := ibc.DefaultChannelOpts()
+	for _, opt := range channelOpts {
+		opt(&channelOptions)
+	}
+
+	eRep := s.getRelayerExecReporter()
+	s.Require().NoError(ic.Build(ctx, eRep, ibctest.InterchainBuildOptions{
+		TestName:          s.T().Name(),
+		HomeDir:           home,
+		Client:            s.Client,
+		NetworkID:         s.network,
+		CreateChannelOpts: channelOptions,
+	}))
+
+	s.startRelayerFn = func(relayer ibc.Relayer) {
+		err := relayer.StartRelayer(ctx, eRep, pathName)
+		s.Require().NoError(err, fmt.Sprintf("failed to start relayer: %s", err))
+		s.T().Cleanup(func() {
+			if !s.T().Failed() {
+				if err := relayer.StopRelayer(ctx, eRep); err != nil {
+					s.T().Logf("error stopping relayer: %v", err)
+				}
+			}
+		})
+		// wait for relayer to start.
+		time.Sleep(time.Second * 10)
+	}
+
+	return r
+}
+
+// GetRelayerWallets returns the relayer wallets associated with the source and destination chains.
+func (s *E2ETestSuite) GetRelayerWallets(relayer ibc.Relayer) (ibc.RelayerWallet, ibc.RelayerWallet, error) {
+	srcChain, dstChain := s.GetChains()
+	srcRelayWallet, ok := relayer.GetWallet(srcChain.Config().ChainID)
+	if !ok {
+		return ibc.RelayerWallet{}, ibc.RelayerWallet{}, fmt.Errorf("unable to find source chain relayer wallet")
+	}
+	dstRelayWallet, ok := relayer.GetWallet(dstChain.Config().ChainID)
+	if !ok {
+		return ibc.RelayerWallet{}, ibc.RelayerWallet{}, fmt.Errorf("unable to find destination chain relayer wallet")
+	}
+	return srcRelayWallet, dstRelayWallet, nil
+}
+
+// RecoverRelayerWallets adds the corresponding relayer address to the keychain of the chain.
+// This is useful if commands executed on the chains expect the relayer information to present in the keychain.
+func (s *E2ETestSuite) RecoverRelayerWallets(ctx context.Context, relayer ibc.Relayer) error {
+	srcRelayerWallet, dstRelayerWallet, err := s.GetRelayerWallets(relayer)
+	if err != nil {
+		return err
+	}
+
+	srcChain, dstChain := s.GetChains()
+
+	if err := srcChain.RecoverKey(ctx, SourceRelayerName, srcRelayerWallet.Mnemonic); err != nil {
+		return fmt.Errorf("could not recover relayer wallet on source chain: %s", err)
+	}
+	if err := dstChain.RecoverKey(ctx, DestinationRelayerName, dstRelayerWallet.Mnemonic); err != nil {
+		return fmt.Errorf("could not recover relayer wallet on destination chain: %s", err)
+	}
+	return nil
+}
+
+// StartRelayer starts the given relayer.
+func (s *E2ETestSuite) StartRelayer(relayer ibc.Relayer) {
+	if s.startRelayerFn == nil {
+		panic("cannot start relayer before it is created!")
+	}
+	s.startRelayerFn(relayer)
```

Comment:
```suggestion
func (s *E2ETestSuite) StartRelayer(relayer ibc.Relayer) {
	if s.startRelayerFn == nil {
		panic("cannot start relayer before it is created!")
	}
	
	s.startRelayerFn(relayer)
```

I like adding extra spacing in these cases, just to separate the code based on error (in this case a nil startRelayerFn) and the code based on a successful executrion (starting the relayer)